scripts/ci/list-workspaces-with-changes.js

```diff
@@ -1,14 +1,64 @@
-import { GitUtils } from "@backstage/cli-node";
+import { execFile as execFileCb } from 'child_process';
+import { promises as fs } from 'fs';
+import { promisify } from 'util';
+import { resolve as resolvePath } from 'path';
+import { EOL } from 'os';
 
-const changedFiles = await GitUtils.listChangedFiles("origin/main");
-const workspaces = new Set();
+import * as url from 'url';
+const __dirname = url.fileURLToPath(new URL('.', import.meta.url));
 
-changedFiles.forEach((name) => {
-  // First match starting with prefix = "workspaces/" & suffix = "/"
-  const match = name.match("(?<=workspaces/)([^/]+)(?=/)");
-  if (match && match[0]) {
-    workspaces.add(match[0]);
+const parentRef = process.env.COMMIT_SHA_BEFORE || 'origin/main';
+
+const execFile = promisify(execFileCb);
+
+async function runPlain(cmd, ...args) {
+  try {
+    const { stdout } = await execFile(cmd, args, { shell: true });
+    return stdout.trim();
+  } catch (error) {
+    if (error.stderr) {
+      process.stderr.write(error.stderr);
+    }
+    if (!error.code) {
+      throw error;
+    }
+    throw new Error(
+      `Command '${[cmd, ...args].join(' ')}' failed with code ${error.code}`,
+    );
   }
-});
+}
+
+async function main() {
+  if (!process.env.GITHUB_OUTPUT) {
+    throw new Error('GITHUB_OUTPUT environment variable not set');
+  }
+
+  const repoRoot = resolvePath(__dirname, '..', '..');
```

Comment:
This could also make sense to inject instead of compute. Especially if we want to make it reusable

Reply:
Yeah, it could for sure, I think we can refactor some of these scripts both here and in the main repo, there's a lot of re-use of helpers etc too.